pkg/controllermanager/deployer/helm/helm.go

```diff
@@ -232,15 +231,6 @@ func (deployer *Deployer) handleDescription(descCopy *appsapi.Description) error
 			fmt.Sprintf("can not find a ManagedCluster with uid=%s in current namespace", descCopy.Labels[known.ClusterIDLabel]))
 		return fmt.Errorf("failed to find a ManagedCluster declaration in namespace %s", descCopy.Namespace)
 	}
-	if mcls[0].Status.AppPusher == nil {
```

Comment:
If so, there is no need to list ManageClusters. Please remove it as well.

Reply:
done.